File: springboot/src/test/java/com/sincrodev/sidebyside/SprintItemRepositoryTest.java
package com.sincrodev.sidebyside;

import com.sincrodev.sidebyside.domain.SprintItem;
import com.sincrodev.sidebyside.domain.SprintItemRepository;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class SprintItemRepositoryTest {

    @Autowired
    private SprintItemRepository repository;

    @Test
    void persistsAndReadsItemsInOrder() {
        repository.saveAll(List.of(
            new SprintItem(1, "Bootstrap repo",          "infra",   "done",        3),
            new SprintItem(2, "Add Symfony service",     "backend", "done",        5),
            new SprintItem(3, "Add Spring Boot service", "backend", "done",        5),
            new SprintItem(4, "Write README",            "docs",    "in_progress", 2),
            new SprintItem(5, "Publish blog post",       "docs",    "planned",     3)
        ));

        List<SprintItem> items = repository.findAllByOrderByIdAsc();

        assertThat(items).hasSize(5);
        // Orden estable por id (paridad con Symfony)
        assertThat(items.stream().map(SprintItem::getId).toList()).containsExactly(1, 2, 3, 4, 5);
        assertThat(items.get(0).getTitle()).isEqualTo("Bootstrap repo");

        // El contrato de cálculo se mantiene: done=13, total=18 -> 72%
        int totalWeight = items.stream().mapToInt(SprintItem::getWeight).sum();
        int doneWeight = items.stream()
            .filter(i -> "done".equals(i.getStatus()))
            .mapToInt(SprintItem::getWeight)
            .sum();
        assertThat(totalWeight).isEqualTo(18);
        assertThat(doneWeight).isEqualTo(13);
        assertThat(doneWeight * 100 / totalWeight).isEqualTo(72);
    }
}

File: symfony/tests/SprintItemRepositoryTest.php
<?php

namespace App\Tests;

use App\DataFixtures\SprintItemSeed;
use App\Entity\SprintItem;
use App\Repository\SprintItemRepository;
use Doctrine\ORM\EntityManagerInterface;
use Doctrine\ORM\Tools\SchemaTool;
use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;

class SprintItemRepositoryTest extends KernelTestCase
{
    public function testPersistsAndReadsItemsInOrder(): void
    {
        self::bootKernel();
        /** @var EntityManagerInterface $em */
        $em = static::getContainer()->get('test.em');

        $schemaTool = new SchemaTool($em);
        $metadata = $em->getMetadataFactory()->getAllMetadata();
        $schemaTool->dropSchema($metadata);
        $schemaTool->createSchema($metadata);

        foreach (SprintItemSeed::items() as $item) {
            $em->persist($item);
        }
        $em->flush();
        $em->clear();

        /** @var SprintItemRepository $repository */
        $repository = $em->getRepository(SprintItem::class);
        $items = $repository->findAllOrdered();

        $this->assertCount(5, $items);
        // Orden estable por id (paridad con Spring Boot)
        $this->assertSame([1, 2, 3, 4, 5], array_map(fn ($i) => $i->getId(), $items));
        $this->assertSame('Bootstrap repo', $items[0]->getTitle());

        // El contrato de cálculo se mantiene: done=13, total=18 -> 72%
        $totalWeight = array_sum(array_map(fn ($i) => $i->getWeight(), $items));
        $doneWeight = array_sum(array_map(fn ($i) => 'done' === $i->getStatus() ? $i->getWeight() : 0, $items));
        $this->assertSame(18, $totalWeight);
        $this->assertSame(13, $doneWeight);
        $this->assertSame(72, intdiv($doneWeight * 100, $totalWeight));
    }
}
